UI/CardScrollUI.gd

```diff
@@ -0,0 +1,54 @@
+extends Control
+
+var card_scene: PackedScene = preload("res://Cards/Card.tscn")
+
+var discard_pile: Array[CardBase] = CardManager.card_container.discard_pile
+var draw_pile: Array[CardBase]    = CardManager.card_container.draw_pile
+var deck_pile: Array[CardBase]         = CardManager.card_container.default_deck
+var cards_to_display: Array[CardBase] = []
+var card_worlds: Array[CardWorld] = []
+var cardworld: CardWorld = null
+var cardui: Control = null
+
+const CARD_SCALE: Vector2 = Vector2(.7, .7)
+const CARDUI_INDEX: int = 2
+const CARDUI_POS: Vector2 = Vector2(4,4)
+const SIZE_OFFSET: Vector2 = Vector2(10,10)
+
+func _on_button_pressed() -> void:
+	queue_free()
+
+func _input(_inputevent: InputEvent) -> void:
+	if Input.is_key_pressed(KEY_ESCAPE):
+		queue_free()
+
+
+
+func populate(parent_name: String) -> void:
+	
+	match parent_name:
+		"DiscardPile":
+			cards_to_display = discard_pile
+			$Label.text = "Showing the discard pile"
+		"DrawPile":
+			cards_to_display = draw_pile
+			$Label.text = "Showing the draw pile"
+		"DeckPile":
+			cards_to_display = deck_pile
+			$Label.text = "Showing the deck pile"
+	
+	cards_to_display.sort()
```

Comment:
what are they sorted by ? how does the sort function works for an array of `Cardbase` ?
I don't think the sort function is implemented by default for us, and i didn't see an implementation either. My guess is that it currently sort based on object index, which for the player might not have any logic.

Some things we can do for now:
- add a sort function (call it `card_sorting`)
- use this on the array of cards
- good thing is that now we can change the internal logic of our sorting function without having to touch this code again (we could do alphabetical order, obtained order, by type etc...)

Reply:
The cards cards can only be sorted by title for now, as there's nothing storing their type or a history of card obtained. I will make a function named card_sort_by_title and use  the sort_custom fuction to applied it.